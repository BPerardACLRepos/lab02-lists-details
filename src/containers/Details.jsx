import React from 'react';
import { Link } from 'react-router-dom';
import Character from '../components/character/Character';
import { getCharacter } from '../services/avatarApi';

export default class Details extends React.Component {
    state = {
        loading: true,
        character: {},
    }

    async componentDidMount() {
        const character = await getCharacter(this.props.match.params.id);
        this.setState({
            loading: false,
            character,
        });
    }
    render() {
        const { loading, character } = this.state;
        return (
            <div>
                <Link to="/">
                    List
                </Link>
                {loading &&
                    <h2>Loading...</h2>
                }
                {!loading &&
                    <Character
                        id={character._id}
                        image={character.photoUrl}
                        name={character.name}
                    />
                }
            </div>
        )
    }
}