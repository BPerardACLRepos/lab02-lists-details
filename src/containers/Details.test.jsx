import React from 'react';
import { screen, render } from '@testing-library/react';
import Details from './Details';

describe('Avatar Details Container', () => {
    it('Renders a single Avatar character to the screen', async () => {
        render(<Details
            match={{
                params: { id: '5cf5679a915ecad153ab68e9' }
            }} />)

        screen.getByText('Loading...');

        const name = await screen.findByText('Bolin');

        expect(name).toMatchSnapshot();
    });
});