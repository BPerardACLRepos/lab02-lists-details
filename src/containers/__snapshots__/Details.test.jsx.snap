// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Details Container Renders a single Avatar character to the screen 1`] = `
<h3>
  Bolin
</h3>
`;
